### PR TITLE
Workspace improvements: formatting and dependencies

### PR DESCRIPTION
**Pull request by petebacondarwin**
- Implemented pnpm catalogs rather than syncpack
- Implemented additional prettier config (primarily around imports)

### CODE REVIEW

File: examples/durable-object/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 		"@cloudflare/workers-types": "^4.20240919.0",
```
Comment (jamesopstad) [inline examples/durable-object/package.json]: Should this also come from `catalog`?

File: examples/durable-object/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
 		"@cloudflare/workers-types": "^4.20241004.0",
```
Comment (jamesopstad) [inline examples/durable-object/package.json:9]: Should this also come from `catalog`?

File: packages/vite-plugin-cloudflare/package.json [hunk 3]
```diff
@@ -7,33 +7,33 @@
...
-		"vite": "6.0.0-beta.2",
-		"vitest": "^2.1.2",
+		"typescript": "catalog:default",
+		"vite": "catalog:default",
+		"vitest": "catalog:default",
+		"wrangler": "catalog:default"
+	},
+	"peerDependencies": {
+		"vite": "catalog:default",
 		"wrangler": "^3.80.4"
```
Comment (jamesopstad) [inline packages/vite-plugin-cloudflare/package.json:37]: Should this also come from `catalog`?

File: examples/durable-object/package.json (hunk 1 above)
Comment (petebacondarwin) [reply examples/durable-object/package.json]: Oops...

**Review by jamesopstad — APPROVED** [review]

**Pull request merged**